Sorting/insertionSort.js

```diff
@@ -0,0 +1,52 @@
+//insertion sort works by defining the first element in the array as sorted.
+//then inserting each subsequent element in the array in the correct location, progressively building a sorted array.
+
+
+function insertionSort(arr) {
+  var sortedPortion = arr.slice(0,1)
+  for (var i = 1; i<arr.length; i++) {
+    sortedPortion = insertNumberIntoSortedArray(arr[i],sortedPortion)
+  }
+  return sortedPortion;
+}
+//the helper function below was the trickiest part of this problem. i'm not sure that it's efficient. the flow is all bor
+//given a sorted array and a number input, the function outputs an array with that number in the correct position.
+function insertNumberIntoSortedArray(num, arr) {
+  //if the array is empty, just put the number in it.
+  if (arr.length === 0) {
```

Comment:
I think this is possible to do without any special cases. I believe there is a way to do it so that the code works for `arr.length >= 0`